### ISSUE TITLE
Cannot access downstream k8s APIs with auth providers enabled

### ISSUE THREAD

**Issue by gitlawr** [rancher/rancher#33798]
**Rancher Server Setup**
- Rancher version: master-1adc8ef2dc359e67b683ada2e552722f9b547b98-head
- Installation option (Docker install/Helm Chart): docker run

**Information about the Cluster**
- Kubernetes version: v1.21.3+k3s1
- Cluster Type (Local/Downstream): imported k3s

**Describe the bug**

**To Reproduce**
1. Install rancher using docker.
2. Create a k3s cluster and import it to the rancher.
3. Verify that the default admin can access downstream k8s APIs via `<SERVER_URL>/k8s/clusters/c-some-id/apis`
4. Enable Github authentication provider.
5. Check access to `<SERVER_URL>/k8s/clusters/c-some-id/apis`.

**Result**
It returns 403 forbidden.
```
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {
    
  },
  "status": "Failure",
  "message": "groups \"github_org://some-id\" is forbidden: User \"system:serviceaccount:cattle-impersonation-system:cattle-impersonation-user-ls749\" cannot impersonate resource \"groups\" in API group \"\" at the cluster scope",
  "reason": "Forbidden",
  "details": {
    "name": "github_org://some-id",
    "kind": "groups"
  },
  "code": 403
}
```

**Expected Result**
Be able to access the APIs.

**Additional context**
It also breaks using the AD auth provider.
Access to the local cluster APIs(`<SERVER_URL>/k8s/clusters/local/apis`) works.

**Comment by deniseschannon** [comment]
I beleive this has already been resolved due to another bug that was fixed. @gitlawr Can you re-test it?

**Comment by gitlawr** [comment]
Resolved in v2.6.0-rc4
